src/TransportBuilder.php

```diff
@@ -125,19 +131,24 @@ public function build(): Transport
 
     /**
      * Return the URL of Elastic Cloud from the Cloud ID
+     * 
+     * @throws Exception\CloudIdParseException
      */
     private function parseElasticCloudId(string $cloudId): string
     {
-        try {
-            list($name, $encoded) = explode(':', $cloudId);
-            list($uri, $uuids)    = explode('$', base64_decode($encoded));
-            list($es,)            = explode(':', $uuids);
-
-            return sprintf("https://%s.%s", $es, $uri);
-        } catch (Throwable $t) {
-            throw new Exception\CloudIdParseException(
-                'Cloud ID not valid'
-            );
+        if (strpos($cloudId, ':') !== false) {
+            list($name, $encoded) = explode(':', $cloudId, 2);
+            $base64 = base64_decode($encoded);
```

Comment:
Wouldn't `base64_decode()` raise an exception if `$encoded` is not a valid base64 string? You have removed the try/catch in this version, it seems you'd still need it to catch unexpected errors such as this.

Reply:
`base64_decode` doesn't throw an exception when the string is not base64. It returns `false` but we need to enable the `strict` mode to true. I'm checking with the documentation [here](https://www.php.net/manual/en/function.base64-decode.php) and I'll update the code.